extensions-core/namespace-lookup/src/main/java/io/druid/query/lookup/NamespaceLookupExtractorFactory.java

```diff
@@ -0,0 +1,307 @@
+        }
+      }
+
+      @GET
+      @Produces(MediaType.APPLICATION_JSON)
+      public Response getMap()
+      {
+        try {
+          return Response.ok(getLatest()).build();
+        }
+        catch (ISE e) {
+          return Response.status(Response.Status.NOT_FOUND).entity(ServletResourceUtils.sanitizeException(e)).build();
+        }
+      }
+
+      private Map<String, String> getLatest()
+      {
+        return ((MapLookupExtractor) get()).getMap();
+      }
+    };
+  }
+
+  @VisibleForTesting
+  public NamespaceLookupExtractorFactory(
+      ExtractionNamespace extractionNamespace,
+      NamespaceExtractionCacheManager manager
+  )
+  {
+    this(extractionNamespace, null, false, manager);
+  }
+
+  @Override
+  public boolean start()
+  {
+    final Lock writeLock = startStopSync.writeLock();
+    writeLock.lock();
+    try {
+      if (!started.compareAndSet(false, true)) {
+        LOG.warn("Already started! [%s]", extractorID);
+        return true;
+      }
+      if (!manager.scheduleAndWait(extractorID, extractionNamespace, firstCacheTimeout)) {
+        LOG.warn("Failed to schedule lookup [%s]", extractorID);
+        return false;
+      }
+      LOG.debug("NamespaceLookupExtractorFactory[%s] started", extractorID);
+      return true;
+    }
+    finally {
+      writeLock.unlock();
+    }
+  }
+
+  @Override
+  public boolean close()
+  {
+    final Lock writeLock = startStopSync.writeLock();
+    writeLock.lock();
+    try {
+      if (!started.compareAndSet(true, false)) {
+        LOG.warn("Not started! [%s]", extractorID);
+        return true;
+      }
+      return manager.checkedDelete(extractorID);
+    }
+    finally {
+      writeLock.unlock();
+    }
+  }
+
+  @Override
+  public boolean replaces(@Nullable LookupExtractorFactory other)
+  {
+    if (other != null && other instanceof NamespaceLookupExtractorFactory) {
+      NamespaceLookupExtractorFactory that = (NamespaceLookupExtractorFactory) other;
+      if (isOneToOne() != ((NamespaceLookupExtractorFactory) other).isOneToOne()) {
+        return true;
+      }
+      if (getFirstCacheTimeout() != ((NamespaceLookupExtractorFactory) other).getFirstCacheTimeout()) {
+        return true;
+      }
+      return !extractionNamespace.equals(that.extractionNamespace);
+    }
+    return true;
+  }
+
+  @Override
+  public LookupIntrospectHandler getIntrospectHandler()
+  {
+    return lookupIntrospectHandler;
+  }
+
+  @JsonProperty
+  public ExtractionNamespace getExtractionNamespace()
+  {
+    return extractionNamespace;
+  }
+
+  @JsonProperty
+  public long getFirstCacheTimeout()
+  {
+    return firstCacheTimeout;
+  }
+
+  @JsonProperty
+  public boolean isOneToOne()
+  {
+    return oneToOne;
+  }
+
+  private String buildID()
+  {
+    return UUID.randomUUID().toString();
+  }
+
+  // Grab the latest snapshot from the cache manager
+  @Override
+  public LookupExtractor get()
```

Comment:
Mostly because it uses a MapLookupExtractor.

So as long as MapLookupExtractor works and is tested, this impl does nothing special other than have a reasonable way to calculate the cache key.